### ISSUE TITLE
Checkout: isMainNetworkSite is not a function

### ISSUE THREAD

**Issue by sirreal** [Automattic/wp-calypso#14405]
I've seen this error on ~two~ several occasions, but I have been unable to reproduce it consistently.

#### Steps to reproduce
1. Start at `https://wordpress.com/plans/$site` where $site is a Jetpack site on the Free plan.
2. Click "Premium" upgrade button.
3. Complete purchase.
4. Checkout hangs and `Uncaught TypeError: t.isMainNetworkSite is not a function` error appears in the console.

#### What I expected

Complete checkout and go back to my site.

#### What happened instead

I get stuck on the checkout screen.

#### Browser / OS version

Chrome 60 / macOS

#### Screenshot / Video

![console](https://cloud.githubusercontent.com/assets/841763/26392892/62b5a88a-4068-11e7-9991-503bc0d04d6c.png)

#### Context / Source

I was investigating into another issue (#7228).

#dogfooding, #manual-testing

**Comment by sirreal** [comment]
A quick look shows this line appears to be the culprit:

![error](https://cloud.githubusercontent.com/assets/841763/26393175/5196ba16-4069-11e7-8550-13ab5e4bfc2d.png)


```js
} else if ( selectedSite.is_multisite && ! selectedSite.isMainNetworkSite() ) {
```

https://github.com/Automattic/wp-calypso/blob/b46d89f2ceff4fd717cf7d63bff32ae60641cd9e/client/my-sites/upgrades/checkout-thank-you/jetpack-plan-details.jsx#L44

**Comment by tyxla** [comment]
Fix is coming in #14766 and it could use a review @sirreal if you can spare a couple of minutes.
